app/component/itinerary/navigator/NaviCardContainer.js

```diff
@@ -17,12 +17,21 @@ import {
   LEGTYPE,
 } from './NaviUtils';
 import usePrevious from './hooks/usePrevious';
+import { usePushNotification } from './hooks/usePushNotification';
 
 const COUNT_AT_LEG_END = 2; // update cycles within DESTINATION_RADIUS from leg.to
 const HIDE_TOPCARD_DURATION = 2000; // milliseconds
 
-function addMessages(incomingMessages, newMessages) {
+function addMessages(
+  incomingMessages,
+  newMessages,
+  oldMessages,
+  createNotification,
+) {
   newMessages.forEach(m => {
+    if (oldMessages && !oldMessages.get(m.id)) {
+      createNotification(m.pushNotification.title, m.pushNotification.content);
+    }
     incomingMessages.set(m.id, m);
   });
 }
```

Comment:
I suggest moving this to component function body. Message set and createNotification are conveniently visible there.